R/modify_pd.R

```diff
@@ -9,33 +9,27 @@ NULL
 
 #' @describeIn update_indention Inserts indetion based on round brackets.
 indent_round <- function(pd, indent_by) {
-  indention_needed <- needs_indention(pd, token = "'('")
+  opening <- which(pd$token == "'('")
+  indention_needed <- needs_indention(pd, token = "'('", opening[1])
   if (indention_needed) {
-    opening <- which(pd$token == "'('")
     start <- opening + 1
     stop <- nrow(pd) - 1
-    if (start > stop) return(pd)
-
-    pd <- pd %>%
-      mutate(indent = indent + ifelse(seq_len(nrow(pd)) %in% start:stop,
-                                      indent_by, 0))
+    pd$indent <- pd$indent +
+      ifelse(between(seq_len(nrow(pd)), start, stop), indent_by, 0)
   }
 
   pd %>%
     set_unindention_child(token = "')'", unindent_by = indent_by)
 }
 #' @rdname update_indention
 indent_curly <- function(pd, indent_by) {
-  indention_needed <- needs_indention(pd, token = "'{'")
+  opening <- which(pd$token == "'{'")
+  indention_needed <- needs_indention(pd, token = "'{'", opening[1])
```

Comment:
Ok, sounds good. I think since `dplyr::between()` returns boolean values, we better call it `compute_indent_flags()` and do boolean subsetting instead of integer subsetting. Does that sound reasonable?
```r
indent_round <- function(pd, indent_by) {
  indent_flags <- compute_indent_flags(pd, token = "'('")
  pd$indent[indent_flags] <- pd$indent[indent_flags] + indent_by
  pd %>%
    set_unindention_child(token = "')'", unindent_by = indent_by)
}

compute_indent_flags <- function(pd, token = "'('") {
  npd <- nrow(pd)
  opening <- which(pd$token == token)
  if (!needs_indention(pd, token, opening[1])) return()
  start <- opening + 1
  stop <- npd - 1
  between(seq_len(npd), start, stop)
}
```
It has only slightly worse performance (<1%) so I think we should do it. It also reduces code duplication.